langchain4j/src/test/java/dev/langchain4j/agent/tool/DefaultToolExecutorTest.java

```diff
@@ -296,4 +294,88 @@ public void should_not_execute_tool_with_null_execution_request() throws NoSuchM
                 .isThrownBy(() -> new DefaultToolExecutor(new TestTool(), (ToolExecutionRequest) null));
 
     }
-}
\ No newline at end of file
+
+    private static class PersonTool {
+
+        @Tool
+        public Person save( Person arg) {
+            assert arg != null;
+            return arg;
+        }
+
+        @Tool
+        public Person[] saveArray(Person[] arg) {
+            assert arg != null;
+            assert arg.length == 2;
+            return arg;
+        }
+
+        @Tool
+        public List<Person> saveList(List<Person> personList) {
```

Comment:
This does not work, the list contains maps, not `Person`s

Reply:
My bad, Missed an `else`. Fixed now.